app/models/legal_aid_application.rb

```diff
@@ -92,4 +94,8 @@ def create_app_ref
   def set_open_banking_consent_choice_at
     self.open_banking_consent_choice_at = Time.current if will_save_change_to_open_banking_consent?
   end
+
+  def create_other_assets_declaration
+    OtherAssetsDeclaration.create(legal_aid_application_id: id)
+  end
```

Comment:
`create_other_assets_declaration` should already be created as one of the dynamic methods generated by the `has_one` declaration. See [docs](https://guides.rubyonrails.org/association_basics.html#methods-added-by-has-one)

I'd also suggest that `build_other_assets_declaration` and/or `create_other_assets_declaration` are used in either the controller or the associated form. That way - if this object is removed in the future the dev doing it only has to look in the controller responsible for adding the declarations - rather than looking for `after_create` callbacks on other models.

Reply:
Agreed